File: string_permutation_problem/permute.py
def permute_string(my_str):
    string_list = []
    # This is a recursive solution to the string permutation problem.
    # The user is supposed to return all the permutations for a given string.
    # Base Case
    if len(my_str) == 1:
        string_list = [my_str]
    else:
        # For every letter in String
        for i, letter in enumerate(my_str):
            # for every permutation resulting from Step 2 and 3 described above
            for perm in permute_string(my_str[:i] + my_str[i+1:]):
                # Add it to output
                string_list += [letter + perm]
    return string_list


def main():
    print(permute_string("abc"))


if __name__ == '__main__':
    main()

File: string_permutation_problem/permute.java
public class Permute{

    /* A recursive solution to a classic problem to print all the permutations 
      in a string. */

public static void permutation(String str) { 
    permutation("", str); 
}

private static void permutation(String prefix, String str) {
    int n = str.length();
    if (n == 0) System.out.println(prefix);
    else {
        for (int i = 0; i < n; i++)
            permutation(prefix + str.charAt(i), str.substring(0, i) + str.substring(i+1, n));
    }
}

  
 public static void main(String[] args) {
        permutation("abc");
    		
	 }

}
